model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProvider.java

```diff
@@ -80,16 +80,14 @@ private Function<MapUserSessionEntity, UserSessionModel> userEntityToAdapterFunc
         // Clone entity before returning back, to avoid giving away a reference to the live object to the caller
         return (origEntity) -> {
             if (origEntity == null) return null;
-            if (isExpired(origEntity, false)) {
-                if (TRANSIENT == origEntity.getPersistenceState()) {
+            if (TRANSIENT == origEntity.getPersistenceState()) {
+                // only check expiry for transient entities, as the physical map store will not return expired entities on any query
```

Comment:
Unfortunately, this assumption is not correct. The expiration check for persistent entities has to remain here, see the following javadoc.
https://github.com/keycloak/keycloak/blob/148c7695ff5b32f46f05a5f341ca5456da144987/model/map/src/main/java/org/keycloak/models/map/common/ExpirableEntity.java#L24-L27